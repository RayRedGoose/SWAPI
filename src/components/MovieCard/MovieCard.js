import React from 'react';
import './MovieCard.scss';

const MovieCard = () => {

  return (
    <section className="movie-card">
      <h3></h3>
      <div></div>
      <h4></h4>
      <p></p>
      <button></button>
    </section>
  )
}

export default MovieCard;
