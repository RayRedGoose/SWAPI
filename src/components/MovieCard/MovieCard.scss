.movie-card {
  text-align: center;
  background-color: rgba(255, 255, 255, 0.9);
  background-color: rgba(100, 149, 237, 0.9);
  margin: 0 1em 0 1em;
  padding: 0 2em 1em 2em;
  width: 15em;
}

h3 {
  font-size: 3em;
  border-bottom: 1px black solid;
  font-weight: bold;
  margin-bottom: 0;
  color: #5A2C05;
}

h4 {
  font-size: 2em;
  font-weight: normal;
  padding: 0;
  margin: .5em;
  color: #5A2C05;
}

p {
  font-size: 1.5em;
  padding-bottom: 1.5em;
}

button {
  width: 232px;
  height: 36px;
  left: 351px;
  top: 560px;
  background-color: #F1AA61;
  color: #5A2C05;
  border: 1px solid #964809;
  box-sizing: border-box;
}
