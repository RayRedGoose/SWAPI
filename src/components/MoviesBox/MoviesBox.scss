.movie-box {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  overflow: scroll;
  padding-bottom: 10em;
}
