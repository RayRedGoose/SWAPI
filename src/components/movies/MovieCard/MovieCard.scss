.movie-card {
  text-align: center;
  background-color: rgba(255, 255, 255, 0.9);
  margin: 10px;
  padding: 40px 20px;
  min-width: 15%;
  min-height: 75%;
  h3 {
    font-size: 2.5em;
    border-bottom: 2px solid #5A2C05;
    font-weight: bold;
    margin-bottom: 5%;
    color: #5A2C05;
    min-height: 25%;
    padding-bottom: 5px;
    margin-bottom: 10px;
  }

  h4 {
    font-size: 1.5em;
    font-weight: normal;
    margin: 1%;
    min-height: 30%;
    color: #5A2C05;
  }

  p {
    font-size: 1.2em;
    padding-bottom: 20%;
  }

  button {
    background-color: #F1AA61;
    color: #5A2C05;
    cursor: pointer;
    border: 1px solid #964809;
    box-sizing: border-box;
    font-size: 1.25em;
    padding: 0.4em 1em;
  }
}

.view-chr {
  text-decoration: none;
  color: #5A2C05;
}
