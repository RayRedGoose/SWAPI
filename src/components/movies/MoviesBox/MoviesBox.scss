.movie-box {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  overflow: scroll;
  padding: 0 20px;
  margin: 0 20px;
}
