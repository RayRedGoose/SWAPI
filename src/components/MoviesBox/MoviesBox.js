import React, { Component } from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesBox.scss';

export default class MovieBox extends Component {
  constructor() {
    super()
    this.state = {
      cards: []
    }
  }

  getMovieInfo = () => {
    return null
  }
  generateCards = () => {
    return null
  }

  render() {
    return (
      <section className="movie-box">
        {this.generateCards()}
      </section>
    )
  }
}
