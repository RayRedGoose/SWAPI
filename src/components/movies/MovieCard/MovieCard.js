import React from 'react'
import { NavLink } from 'react-router-dom'
import './MovieCard.scss'

const MovieCard = (props) => {
  const year = props.release_date.split('-')[0]

  return (
    <section className="movie-card">
      <h3>Episode {props.id}</h3>
      <h4>{props.title}</h4>
      <p>{year}</p>

      <button>
        <NavLink
          to={`/movies/${props.id}`}
          key={props.id}
          className="view-chr">View characters</NavLink>
      </button>
    </section>
  )
}

export default MovieCard;
