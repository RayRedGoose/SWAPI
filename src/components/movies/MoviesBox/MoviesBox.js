import React, { Component } from 'react'
import MovieCard from '../MovieCard/MovieCard'
import PropTypes from 'prop-types'
import './MoviesBox.scss'

export default class MovieBox extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  componentDidMount = () => {
    this.props.addMovies()
  }

  generateCards = () => {
    return this.props.movies.map((movie, index) => (
      <MovieCard
        key={index}
        id={index+1}
        {...movie} />
      )
    )
  }

  render() {
    return (
      <section className="movie-box">
        { this.generateCards() }
      </section>
    )
  }
}

MovieBox.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  addMovies: PropTypes.func.isRequired
}
