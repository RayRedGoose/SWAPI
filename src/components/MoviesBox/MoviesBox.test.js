import React from 'react';
import { shallow } from 'enzyme';
import MoviesBox from './MoviesBox';
import MovieCard from '../MovieCard/MovieCard';

describe("MoviesBox", () => {

  it("should render MoviesBox", () => {
    const wrapper = shallow(
      <MoviesBox
      />
    )
  })

  it("should generate cards from state", () => {
    const wrapper = shallow(<MoviesBox />)
    const fakeMovie1 = {
      title: "A Star War",
      episode_id: 4,
      director: "The Rock",
      producer: "Mr. Tumnus",
      release_date: "1977-05-25"
    }
    const fakeMovie2 = {
        title: "The Fast and the Furious",
        episode_id: 3,
        director: "Dominic Toretto",
        producer: "Cars",
        release_date: "2001-05-25"
    }
    const fakeMovie3 = {
        title: "Bambi",
        episode_id: 2,
        director: "Michael Bay",
        producer: "Somebody Nobody",
        release_date: "2005-05-25"
    }
    wrapper.setState({cards: [fakeMovie1, fakeMovie2, fakeMovie3]})
    const expected1 = ( <MovieCard
        title = "Bambi"
        episode = "2"
        year = "2005"
        /> )
    const expected2 = ( <MovieCard
        title = "The Fast and the Furious"
        episode = "3"
        year = "2001"
        /> );

      expect(wrapper.instance().generateCards().length).toEqual(3);
      expect(wrapper.instance().generateCards()[2].props.title).toEqual(expected1.props.title);

  })
})
