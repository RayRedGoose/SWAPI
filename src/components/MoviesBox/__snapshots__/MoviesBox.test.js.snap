// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesBox should render MoviesBox 1`] = `
<section
  className="movie-box"
/>
`;
